substrate/frame/assets/src/lib.rs

```diff
@@ -495,6 +495,10 @@ pub mod pallet {
 		Created { asset_id: T::AssetId, creator: T::AccountId, owner: T::AccountId },
 		/// Some assets were issued.
 		Issued { asset_id: T::AssetId, owner: T::AccountId, amount: T::Balance },
+		/// Some assets were deposited (e.g. for transaction fees).
+		Deposited { asset_id: T::AssetId, who: T::AccountId, amount: T::Balance },
+		/// Some assets were withdrawn from the account (e.g. for transaction fees).
+		Withdrawn { asset_id: T::AssetId, who: T::AccountId, amount: T::Balance },
```

Comment:
Should these be added to the end of the enum? Technically this is a breaking change.

Reply:
I guess not, since decoding applies on each code update AFAIK. However, I just moved them to the end of the enum, it makes sense to reduce breaking as much as possible.

Reply:
> since decoding applies on each code update

What do you mean here? Sure in the runtime, but there are external libraries that depend on the variant encoding.

Reply:
Yes, you're right! I agree reducing breakage from end-to-end is necessary. I ended up moving those variants to the end of the enum to ensure they don't affect runtime users.